Upgrade CTST deps

williamlardier opened the pull request:
Bump CTST dependencies

Also, adapt the code to follow the scality rules, as well as the typescript recommended rules.

Seragonia committed:
Bump eslint to v9

- Remove unused dependencies
- Use scality eslint rules
- Use recommended typescript eslint rules

Issue: ZENKO-4882

Seragonia committed:
Bump dependencies

Issue: ZENKO-4882

Seragonia committed:
Update code to match linter rules

Issue: ZENKO-4882

bert-e commented:
# Hello williamlardier,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).





<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4882 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.0`

* `2.6.67`

* `2.7.63`

* `2.8.43`

* `2.9.18`


Please check the `Fix Version/s` of ZENKO-4882, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4882 contains:


* `2.10.0`

* `2.8.43`

* `2.9.18`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.0`

* `2.6.67`

* `2.7.63`

* `2.8.43`

* `2.9.18`


Please check the `Fix Version/s` of ZENKO-4882, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4882 contains:


* `2.10.0`

* `2.7.63`

* `2.8.43`

* `2.9.18`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.0`

* `2.6.67`

* `2.7.63`

* `2.8.43`

* `2.9.18`


Please check the `Fix Version/s` of ZENKO-4882, or the target
branch of this pull request.

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

benzekrimaha commented on tests/ctst/package.json:
```diff
@@ -23,11 +23,10 @@
-    "babel-jest": "^29.3.1",
-    "cli-testing": "github:scality/cli-testing.git#v1.2.0",
-    "eslint": "^8.28.0"
+    "cli-testing": "github:scality/cli-testing.git#0744e229cb8c3dc19eb2f5bbb603c3ba0f583aa0",
```
reminder to put a valid tag before merging

francoisferrand commented on tests/ctst/eslint.config.mjs:
```diff
@@ -0,0 +1,35 @@
+    ...tseslint.configs.recommended,
+    {
+        ignores: [
+            '**/.vscode',
+            '**/vendor/*.js',
+            '**/build',
+            '**/coverage',
+            '**/dist',
+            '**/node_modules/**/*',
+            '**/package',
+            '**/reports',
+            '**/eslint.config.mjs',
```
instead of replicating what is (in large part) in `.gitignore`, should "import" this file with `includeIgnoreFile(gitignorePath)` [1]

[1] https://eslint.org/docs/latest/use/configure/ignore#including-gitignore-files

francoisferrand commented on tests/ctst/features/resource-policies/regen.js:
```diff
@@ -152,21 +152,26 @@ const output = scenarios.map(scenario => {
     const paddedBucketPolicyApplies = scenario.bucketPolicyApplies.padEnd(longest.bucketPolicyApplies);
     const paddedBucketPolicyEffect = scenario.bucketPolicyEffect.padEnd(longest.bucketPolicyEffect);
 
+    // eslint-disable-next-line max-len
```
shouldn't we reformat instead:

```suggestion
return (   
	`           `, paddedAction, paddedBucketPolicyExists. paddedBucketPolicyApplies,
	paddedBucketPolicyEffect, paddedIamPolicyExists, paddedIamPolicyApplies, paddedIamPolicyEffect,
).join(' | )
```

francoisferrand commented on tests/ctst/tsconfig.json:
```diff
@@ -26,6 +26,7 @@
       // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
       "noImplicitThis": true,                   /* Raise error on 'this' expressions with an implied 'any' type. */
       // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
+      "skipLibCheck": true,
```
do we really want to save compilation time at the expense of type-system accuracy?

williamlardier replied:
It is because of issues with some libs we do not control, e.g.:

```
node_modules/cli-testing/node_modules/@aws-sdk/client-s3/dist-types/auth/httpAuthSchemeProvider.d.ts:1:10 - error TS2724: '"@aws-sdk/core"' has no exported member named 'AwsSdkSigV4AAuthInputConfig'. Did you mean 'AwsSdkSigV4AuthInputConfig'?
```

And one vaultclient issue (ticket is VLTCLT-52):
```
node_modules/vaultclient/lib/IAMClient.d.ts:266:100 - error TS1016: A required parameter cannot follow an optional parameter.

266     verifySignatureV4(stringToSign: string, signature: string, accessKey: string, region?: string, scopeDate: string, options: {
```

I will try to remove this line when the sdk is more up to date.

Seragonia committed:
Improve linting

- Ignore files through the gitignore file
- Avoid ignoring eslint rule when we can improve the code

Issue: ZENKO-4882

ghost commented:
/create_integration_branches

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.7/improvement/ZENKO-4882` with contents from `improvement/ZENKO-4882`
and `development/2.7`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.7/improvement/ZENKO-4882 origin/development/2.7
 $ git merge origin/improvement/ZENKO-4882
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.7/improvement/ZENKO-4882
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/ZENKO-4882` into
`development/2.6`
* [w/2.7/improvement/ZENKO-4882](https://github.com/scality/zenko/compare/development/2.7...w/2.7/improvement/ZENKO-4882) will be merged into `development/2.7` 
* [w/2.8/improvement/ZENKO-4882](https://github.com/scality/zenko/compare/development/2.8...w/2.8/improvement/ZENKO-4882) will be merged into `development/2.8` 
* [w/2.9/improvement/ZENKO-4882](https://github.com/scality/zenko/compare/development/2.9...w/2.9/improvement/ZENKO-4882) will be merged into `development/2.9` 
* [w/2.10/improvement/ZENKO-4882](https://github.com/scality/zenko/compare/development/2.10...w/2.10/improvement/ZENKO-4882) will be merged into `development/2.10` 



The following branches will **NOT** be impacted:

* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

williamlardier commented:
@bert-e ?

ghost commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`

* :heavy_check_mark: `development/2.7`

* :heavy_check_mark: `development/2.8`

* :heavy_check_mark: `development/2.9`

* :heavy_check_mark: `development/2.10`



The following branches have **NOT** changed:

* `development/2.5`



Please check the status of the associated issue ZENKO-4882.

Goodbye williamlardier.




*The following options are set:* **approve, create_integration_branches**
